frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.tsx

```diff
@@ -34,9 +36,20 @@ interface DispatchProps {
 
 type Props = MainNavbarProps & StateProps & DispatchProps;
 
-function mapStateToProps(state: State) {
+function getModelDetailPageCard(state: State, params: { slug?: string }) {
+  const entityId = Urls.extractEntityId(params.slug);
+  return Questions.selectors.getObject(state, { entityId });
```

Comment:
This pattern should be discouraged and I believe it was discussed at one of the FE meetings. If an entity is needed, it's preferred to get it via the entity loader, like this:
```
Questions.load({
  id: (state, ({ location, params })) =>  isModelDetailPathname(location) 
   ? Urls.extractEntityId(params.slug)
   : undefined,
entityAlias: "whatever", // choose the name of the prop
loadingAndErrorWrapper: false, // if needed
})
})
```

Reply:
I usually alias cards loaded via `Questions.load` to something that ends with `card` because otherwise it is called `question`, which is incorrect.